clash-lib/src/Clash/Netlist/Util.hs

```diff
@@ -129,9 +129,9 @@ splitNormalized
   -> Term
   -> (Either String ([Id],[LetBinding],Id))
 splitNormalized tcm expr = case collectBndrs expr of
-  (args,Letrec xes e)
+  (args, collectTicks -> (Letrec xes e, ticks))
```

Comment:
My understanding is: ticks can wrap _any_ term, so any time you match on a term you should make sure that

* it either has no ticks
* or to call `collectTicks`

This seems like an impossible thing to do correctly, so ideally I'd like to see a solution where the compiler forces us to handle ticks. Could we think about this please? (Not for this PR though.)